src/main/java/io/nats/client/NatsThread.java

```diff
@@ -33,7 +29,7 @@ public NatsThread(Runnable runnable, String name) {
         super(runnable, name + "-" + created.incrementAndGet());
         setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
             public void uncaughtException(Thread t, Throwable e) {
-                logger.debug("UNCAUGHT in thread {}", t.getName(), e);
+               e.printStackTrace();
```

Comment:
It depends on the context/usage of the NATS thread intended by the caller, but I think most of the time we'll want to keep the thread alive.  I'm also wondering if the NATS thread class is really required.

There are a number of places in the code I'd like to revisit; I've added a few TODOs.